modules/appeals_api/spec/models/supplemental_claim_spec.rb

```diff
@@ -25,6 +25,39 @@
     end
   end
 
+  describe 'before hooks' do
+    before { @supplemental_claim = build(:extra_supplemental_claim) }
+
+    describe 'assign_metadata' do
+      it 'assigns all metadata fields when pact act boolean feature flag is enabled' do
+        Flipper.enable(:decision_review_sc_pact_act_boolean)
+
+        @supplemental_claim.save
+
+        expect(@supplemental_claim.metadata.dig('form_data', 'evidence_type')).not_to be_nil
```

Comment:
Since our fixtures make these values deterministic, let's actually check that their values are what we expect them to be, rather than just a "not nil" check.

This test section on checking metadata also feels like it's going to be a contender for making sure different `api_version`s can store their differing schemas into the same metadata keys.

Reply:
Good call. I'll make that change. 👍

Reply:
Done! ✅ 

@cilestin Could you also take a look at the questions I had under `Requested Feedback`, if you haven't already? Those are items that I felt like I needed another eye/opinion on. Thanks!